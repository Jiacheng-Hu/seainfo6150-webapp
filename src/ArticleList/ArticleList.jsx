import React from "react";
import ArticleListItem from "../ArticleListItem/ArticleListItem";
import PropTypes from "prop-types";

const ArticleList = props => {
	let content;
	if(props.articles.length){
		content = (
			<>
				<ul>{props.articles.map((item) => (
					<ArticleListItem article = {item} key = {item.slug}/>
				))}
				</ul>
			</>
		)
	}else{
		content = <div>You have no data!</div>
	}
	return props.articles.length ? (
		<div>{content}</div>
	) : null;
};

ArticleList.propTypes = {
	articles: PropTypes.arrayOf(
		PropTypes.shape({
			title: PropTypes.string.isRequired,
			displayDate: PropTypes.string.isRequired,
			shortText: PropTypes.string.isRequired
		})
	).isRequired
};

export default ArticleList;