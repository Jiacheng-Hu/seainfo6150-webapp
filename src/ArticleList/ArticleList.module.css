@media screen and (min-width: 360px) {
    .container {
        display: grid;
        grid-template-columns: 1fr;
        flex-direction: column;
        background-color: #808080;
        box-shadow: 0 0 5px 5px rgba(255, 0, 0, 0.5);
        margin: 15px;
        padding-right: 20px;
        padding-bottom: 20px;
        border-radius: 20px;
    }
}

@media screen and (min-width: 540px) {
    .container {
        display: grid;
        grid-template-columns: 1fr 1fr;
        flex-direction: column;
        background-color: #808080;
        box-shadow: 0 0 5px 5px rgba(255, 0, 0, 0.5);
        margin: 15px;
        padding-right: 20px;
        padding-bottom: 20px;
        border-radius: 20px;
    }
}

@media screen and (min-width: 768px) {
    .container {
        display: grid;
        grid-template-columns: 1fr 1fr 1fr;
        flex-direction: column;
        background-color: #808080;
        box-shadow: 0 0 5px 5px rgba(255, 0, 0, 0.5);
        margin: 15px;
        padding-right: 20px;
        padding-bottom: 20px;
        border-radius: 20px;
    }
}

@media screen and (min-width: 1080px) {
    .container {
        display: grid;
        grid-template-columns: 1fr 1fr 1fr 1fr;
        flex-direction: column;
        background-color: #808080;
        box-shadow: 0 0 5px 5px rgba(255, 0, 0, 0.5);
        margin: 15px;
        padding-right: 20px;
        padding-bottom: 20px;
        border-radius: 20px;
    }
}