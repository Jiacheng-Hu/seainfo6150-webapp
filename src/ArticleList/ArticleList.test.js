import React from "react";
import ArticleList from "./ArticleList";

describe("ArticleList tests", () => {
    it("renders correctly", () => {
        const articles = [
            {
                "displayDate": "November 21st 2018, 2:57 pm",
                "author": "Cathryn Lindsey",
                "title": "You can take a pie on a plane",
                "shortText": "Getting excited to eat your mother's roast turkey, your uncle's stuffing and your sister's famous green bean casserole?"
            },
            {
                "displayDate": "November 22nd 2018, 6:58 am",
                "author": "Britney Collier",
                "title": "Boeing's 777X jetliner comes together ",
                "shortText": "Travelers around the globe are one step closer to seeing a plane with its wings folded up being taxied to their gate."
            }
        ];
        const { container } = render(<ArticleList articles = {articles}/>);
        expect(container ).toMatchSnapshot();
    });
})