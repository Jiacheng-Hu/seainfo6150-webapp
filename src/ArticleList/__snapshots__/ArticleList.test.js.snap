// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList tests renders correctly 1`] = `
<div>
  <div>
    <div
      class="container"
    >
      <div
        class="container"
      >
        <div
          class="title"
        >
          You can take a pie on a plane
        </div>
        <button
          class="button"
        >
          Show more
        </button>
      </div>
      <div
        class="container"
      >
        <div
          class="title"
        >
          Boeing's 777X jetliner comes together 
        </div>
        <button
          class="button"
        >
          Show more
        </button>
      </div>
    </div>
  </div>
</div>
`;
